### PR TITLE
feat(menu): coercing overlapTrigger attribute

### PR DESCRIPTION
**Pull request by ocarreterom**
Uses coerceBooleanProperty for `overlapTrigger` attribute.
```
<md-menu overlapTrigger="false"></md-menu>
```

### CODE REVIEW

**Comment by ocarreterom** [comment]
I signed it!

File: src/lib/menu/menu-directive.ts [hunk 1]
```diff
@@ -75,7 +75,10 @@ export class MdMenu implements AfterContentInit, MdMenuPanel, OnDestroy {
   @ContentChildren(MdMenuItem) items: QueryList<MdMenuItem>;
 
   /** Whether the menu should overlap its trigger. */
-  @Input() overlapTrigger = true;
+  @Input()
+  get overlapTrigger(): boolean { return this._overlapTrigger; }
+  set overlapTrigger(value: boolean) { this._overlapTrigger = coerceBooleanProperty(value); }
```
Comment (jelbourn) [inline src/lib/menu/menu-directive.ts]: `coerceBooleanProperty` should only be used for things that start as `false` and are optionally turned to `true`. 

@kara was `overlapTrigger` meant to be set to `true` by default?

Comment (kara) [inline src/lib/menu/menu-directive.ts]: Yes, it should overlap the trigger by default

**Comment by jelbourn** [comment]
This will be captured in #7283

**Pull request closed without merging**